lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1684,65 +1684,69 @@ def plot_surface(self, X, Y, Z, *args, **kwargs):
         if shade and cmap is not None and fcolors is not None:
             fcolors = self._shade_colors_lightsource(Z, cmap, lightsource)
 
+        def boundary_edge(arr):
+            """
+            Get the boundary elements of the rectangle ``arr``,
```

Comment:
Happy to rename / extract this function if someone else can suggest a name / location.

Reply:
heh, I might call it "nibble()", because it nibbles around the edge of the 2D array. At the very least, I would expand the explanation a bit to get that point across. It took me a few minutes to realize that was what was done.

Reply:
Not a fan of `nibble`. How about one of

[`array_`\](`perimeter`|`edge`|`boundary`|`border`)[`_`(`elements`|`values`)]

I think I like `array_perimeter` most.

Can you suggest a top-level place to put this in?

Reply:
Moved to `cbook`, and added an example